File: 6th_Semester/NET_Centric_Computing/Class codes/Basics/Constructor.cs
/*
 * Constructor is a special method that is called when an object is created.
 * It is used to initialize the object, to assign values to the object's fields or properties.
 * The constructor has the same name as the class and does not have a return type.
 * They can be overloaded and a class can have multiple constructors.
 * There are two types of constructors:
 * Parameterized Constructor: Constructor with parameters.
 * Default Constructor: Constructor without parameters.
 */
using System;

namespace Basics
{
    class Constructor
    {
        private string message;

        // Default Constructor
        public Constructor()
        {
            // set message to default value
            this.message = "Default Message";
            Console.WriteLine("Default Constructor");
        }

        // Parameterized Constructor
        public Constructor(string message)
        {
            this.message = message;
        }

        // display method
        public void Display()
        {
            Console.WriteLine("Message: " + message);
        }
    }
}

File: 7th_Semester/Advanced Java Programming/Class Codes/src/main/java/com/suyash/Constructor.java
package com.suyash;

/**
 * Constructor is a special type of method that is used to initialize the object.
 * It is called when an object is created.
 */
public class Constructor {
    // default constructor
    public Constructor() {
        System.out.println("Default constructor");
    }

    // parameterized constructor
    public Constructor(int a, int b) {
        System.out.println("Parameterized constructor");
        System.out.println("Value of a: " + a);
        System.out.println("Value of b: " + b);
    }

    public static void main(String[] args) {
        Constructor c = new Constructor();
        Constructor c1 = new Constructor(10, 20);
    }
}
